trader/trader.go

```diff
@@ -534,3 +614,31 @@ func (t *Trader) getExistingOffers() ([]hProtocol.Offer /*sellingAOffers*/, []hP
 func (t *Trader) setExistingOffers(sellingAOffers []hProtocol.Offer, buyingAOffers []hProtocol.Offer) {
 	t.sellingAOffers, t.buyingAOffers = sellingAOffers, buyingAOffers
 }
+
+func (t *Trader) countOfferChangeTypes(offers []txnbuild.Operation) (numCreate int, numUpdate int, numDelete int, e error) {
+	var offerXdr xdr.Operation
+	for _, o := range offers {
+		offerXdr, e = o.BuildXDR()
+		if e != nil {
+			return 0, 0, 0, fmt.Errorf("could not convert op to offer: %s", e)
+		}
+
+		manageSellOp := offerXdr.Body.ManageSellOfferOp
+		if manageSellOp == nil {
+			return 0, 0, 0, fmt.Errorf("offers were not of expected type ManageSellOffer")
+		}
+
+		// 0 amount represents deletion
+		// 0 offer id represents creating a new offer
+		// anything else represents updating an extiing offer
+		if manageSellOp.Amount == 0 {
+			numDelete++
+		} else if manageSellOp.OfferId == 0 {
+			numCreate++
+		} else {
+			numUpdate++
+		}
+	}
+
+	return
```

Comment:
code convention: we try to avoid the implicit return syntax to keep things simple by being explicit.

can you `return numCreate, numUpdate, numDelete`?